### ISSUE TITLE
New suggestion request for: `f . uncurry (zipWith g) $ (a, b)`

### ISSUE THREAD

**Issue by Javran** [ndmitchell/hlint#1218]
This looks like a good one to have in general:

```haskell
module Z where
lhs f g a b = f . uncurry (zipWith g) $ (a, b)
rhs f g a b = f $ zipWith g a b
```
even GHC think they are the same on -O2:
```sh
$ ghc -O2 Z.hs -ddump-simpl
[1 of 1] Compiling Z                ( Z.hs, Z.o )

==================== Tidy Core ====================
<omitted>
rhs [InlPrag=NOUSERINLINE[2]]
  :: forall c t a b. ([c] -> t) -> (a -> b -> c) -> [a] -> [b] -> t
<omitted>
rhs = lhs
<omitted>
```

**Comment by ndmitchell** [comment]
Agreed! I think the hint should be `uncurry f (a, b) ==> f a b`. It's probably quite easy to add to https://github.com/ndmitchell/hlint/blob/master/data/hlint.yaml (there are a bunch of curry/uncurry hints in there already). PR most welcome!

**Comment by Javran** [comment]
Yeah I didn't attempt to simplify it as I assumed such a simple rule must have existed, turn out it's not:

```
$ cat >Z.hs
module Z where f g x y = uncurry g (x,y)
$ hlint Z.hs
No hints
```

I'll take a look, feel free to assign me on this.

**Comment by ndmitchell** [comment]
Sure, assigned to you :)

If the simple one works, it automatically gets generalised in a bunch of interesting ways, so if you write the simple one, it will catch the more complex example you initially wrote.
